core/src/main/scala/com/clairvoyant/restonomer/core/app/RestonomerContext.scala

```diff
@@ -30,21 +30,34 @@ class RestonomerContext(val restonomerContextDirectoryPath: String) {
 
   val configVariables: Map[String, String] = loadConfigVariables(CONFIG_VARIABLES_FILE_PATH)
 
-  val checkpoints: List[CheckpointConfig] = loadConfigsFromDirectory[CheckpointConfig](
-    configDirectoryPath = CHECKPOINTS_CONFIG_DIRECTORY_PATH,
-    configVariablesSubstitutor = ConfigVariablesSubstitutor(configVariables = configVariables)
-  )
-
-  def runCheckpoint(checkpointName: String): Unit =
-    checkpoints
-      .find(_.name == checkpointName) match {
-      case Some(checkpointConfig) =>
-        runCheckpoint(checkpointConfig)
-      case None =>
-        throw new RestonomerContextException(s"The checkpoint: $checkpointName does not exists.")
+  def runCheckpoint(checkpointName: String): Unit = {
+    val checkpoints: List[CheckpointConfig] = loadConfigsFromDirectory[CheckpointConfig](
+      configDirectoryPath = CHECKPOINTS_CONFIG_DIRECTORY_PATH,
+      configVariablesSubstitutor = ConfigVariablesSubstitutor(configVariables = configVariables),
+      fileName = Option(checkpointName)
+    )
```

Comment:
Will suggest one extra line after this in order to make code look cleaner. Sorry but I focus much on code presentation. So pls bear with me. 🙂

Reply:
resolved